tftest.py

```diff
@@ -343,10 +348,23 @@ def setup(self, extra_files=None, plugin_dir=None, init_vars=None,
       init_vars: Terraform backend configuration variables
       backend: Terraform backend argument
       cleanup_on_exit: remove .terraform and terraform.tfstate files on exit
+      disable_prevent_destroy: set all prevent destroy to false
 
     Returns:
       Terraform init output.
     """
+    # remove lifecycle prevent destroy
+    if disable_prevent_destroy:
+      min_python = (3, 5)
+      if sys.version_info < min_python:
+        raise TerraformTestError('The disable_prevent_destroy flag requires at least Python 3.5')
+      for tf_file in Path(self.tfdir).rglob('*.tf'):
```

Comment:
Do you think it's worth trapping potential file access errors here, and returning a meaningful message?

Reply:
I did consider wrapping this in try/except for file access type issues, but I thought the script should fail if it couldn't carry out the required actions and the exception raised should be meaningful.

If you think it makes the failure easier to understand I'm happy to add this in, were you thinking something like this:


```python
        try:
          shutil.copy(str(tf_file), f'{str(tf_file)}.bkp')
        except OSError as exc:
          _LOGGER.exception(f'Unable to backup terraform file {tf_file.resolve()}')
          raise TerraformTestError(f'Backup of terraform file ({tf_file.resolve()}) failed')
```

Reply:
I just prefer seeing a meaningful message, and raising exceptions that use or extend the same base class, makes it easier to read and to consume. But it's not super important, feel free of course to merge as-is, we can revisit later.

Reply:
I'd rather do it now, not super hard. Let me update the PR.

Reply:
Pushed the updates let e know if that works.